### PR TITLE
Update `@wordpress/components` to v19.8.5

### LINKED ISSUE #6054: Update bundled `@wordpress/components` dependency to version 19.8.5
### Description

We want to update the [@wordpress/components](https://www.npmjs.com/package/@wordpress/components) package to a more recent version, the one being used by our minimum supported version of WordPress:

- Current version: [11.1.5](https://www.npmjs.com/package/@wordpress/components/v/11.1.5)
- Target version: [19.8.5](https://www.npmjs.com/package/@wordpress/components/v/19.8.5)

### Acceptance criteria

- All CI checks are passing
- Package-specific tests are passing 
  - To figure out some of them, it requires researching what this package does, the breaking changes between the two versions, if any, etc...
  - Everywhere we're using the package, the UI still works as expected
  - For TypeScript libraries, ensure we're using the built-in types shipped with the target version instead of our custom types, if applicable
  - If it's too much or it seems too complicated because of a lack of expertise in the relevant frontend areas, feel free to document what's missing to be confident about the update and ask for help from [the team owning this area](https://wootransactp2.wordpress.com/welcome-to-transact/product-responsibilities/)
- The review process from the [dependencies doc](https://github.com/Automattic/woocommerce-payments/blob/develop/docs/dependencies.md#javascript-dev-dependencies) was followed and now has an entry for this package with all the relevant details for future iterations

### Dev notes

Run `npm install --save-exact <packageName@version>` to update a package by an exact version.

This dependency is bundled when we build WCPay, so updating the version locally impacts the version used when using WCPay.

### PR DESCRIPTION
**Pull request by jbordonado**
Fixes #6054 

#### Changes proposed in this Pull Request

- Update the version of `@wordpress/components` to 19.8.5
   - I noticed we relied on `is-size-small`, `is-size-medium` or `is-size-large` CSS classes to override some styles for the `<CardBody>` component especially. Since those CSS classes are no longer used with the new package version, some styles were not applied, so I had to update the CSS in different places. Generally, we specified the `is-size-medium` class (the default size if no `size` prop is passed to the component) to ensure we're not altering styles for other `<CardBody>` of different sizes. Still, I found that removing the `is-size-medium` is enough to fix the problems and is safe because I didn't find places where we would have several `<CardBody>` of different sizes used under the same parent element. So I didn't bother making things more complicated by adding new CSS classes to be able to have enough specificity in the different CSS selectors to override safely
- Update the version of `@types/wordpress__components` to 19.10.5
  - I tried to remove the types dependency, but we cannot use the built-in types. It complains about a missing declaration file. I searched on Slack and saw similar discussions. We need to keep using the DefinitelyTypes types.
- Add an entry in the dependencies doc
#### Testing instructions

- JS tests are passing
- E2E tests are passing
- All flows using components from the package look and feel the same as before

Site that can be used to compare, if needed:
- Regular JN site used for critical flows (currently WCPay 5.9.0-test-2): https://accused-hoopoe.jurassic.ninja/wp-admin/  - `demo` / `tDzp5fxL4EnO`

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6162` or branch name `update/wordpress-components-dependency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/wordpress-components-dependency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: cabb0ae196d9a960d32834833b2b0be0c5e0c3f4
- Build time: 2023-05-23 07:34:55 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: webpack/shared.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ module.exports = {
 	},
 	plugins: [
 		new ProvidePlugin( {
-			process: 'process/browser',
+			process: 'process/browser.js',
```
Comment (jbordonado) [inline webpack/shared.js:129]: To be checked. I'm not sure if it's the right way or what's the impact of this change, though it fixed the errors I was seeing when webpack was compiling through the `npm run dev` command

File: client/multi-currency/multi-currency-settings/enabled-currencies-list/test/__snapshots__/index.js.snap
Comment (jbordonado) [inline client/multi-currency/multi-currency-settings/enabled-currencies-list/test/__snapshots__/index.js.snap:1]: I went on `/wp-admin/admin.php?page=wc-settings&tab=wcpay_multi_currency` and compared the UI for the old version VS the new version. The UI was still fine, and even better than with the previous version.

There remains one failing test in `client/multi-currency/multi-currency-settings/enabled-currencies-list/test/index.js` and it seems like a random ID is generated on each render and used as part of the CSS classes on a certain element. We might need to abandon the snapshot testing for that one if we cannot prevent that behaviour in the context of the test.

**Comment by jbordonado** [comment]
I asked support from different teams to double-check things I'm not too familiar with or not sure how to test easily:
- Sigma (thread: p1683211846016709-slack-C01B8KNUYSW) ; They have a meetup next week, so they can probably review the PR starting May 15
- Moltres (thread: p1683212070845319-slack-C03KTTK2YMA) ✅ 
- Fusion (thread: p1683213328413749-slack-C04DN7Q51KK)
- Helix (thread: p1683213144786849-slack-C02BW3Z8SHK)

I updated the snapshots for the UI parts I could check, which seemed OK, but that still leaves 99 failures.

**Comment by jbordonado** [comment]
Latest status: 

```
Test Suites: 25 failed, 214 passed, 239 total
Tests:       65 failed, 1459 passed, 1524 total
Snapshots:   65 failed, 315 passed, 380 total
```

What's left to check/update:

- (SIGMA) Fraud protection + multi-currency + VAT = 20 files
- (HELIX) Dispute + deposits = 3 files

**Comment by Jinksi** [comment]
In d024b45, I've adjusted the BannerNotice icon margin to fix a change that has occurred with this component update.

Before (5.9.0-develop):
5 9 0-dev – BannerNotice

This PR:
After unfixed – BannerNotice

This PR after style fix in d024b45:
After fixed – BannerNotice

**Review by Jinksi — APPROVED** [review]
Thanks for tackling this @jbordonado 🙌 

I've tested Helix product areas that are impacted by this update including Disputes and Deposits.

**Note**: I'm approving based on the Helix product areas I've looked at. I haven't tested other product areas in detail.

I've updated failing snapshots after confirming that there are no significant changes or issues for: 
* Account Balances Card in 8540f1a
* Dispute Evidence Form in 4f435f1

As mentioned https://github.com/Automattic/woocommerce-payments/pull/6162#issuecomment-1548888709, I've adjusted the BannerNotice icon margin in d024b45.

<details>
<summary>Before/After screenshots of deposits & dispute screens</summary>


### Account overview – Deposits Card

`develop` branch:  
5 9 0-dev – Deposits Card


This branch:  
After fixed – Deposits Card


### Account overview – Instant Deposit Modal

Note: it looks to me like there is a padding issue with this modal. I think this is outside of the scope of this PR, as this existed prior to these changes.

`develop` branch:  
5 9 0-dev – Instant Deposit Modal


This branch:  
After– Instant Deposit Modal

### Deposit Information Timeline


`develop` branch:  
5 9 0-dev – Deposit Timeline



This branch:  
After – Deposit Timeline



### Dispute Details – Lost

`develop` branch:  
5 9 0-dev – Dispute Details Lost


This branch:  
After – Dispute Details Lost

### Dispute Details – Won

`develop` branch:  
5 9 0-dev – Dispute Details Won


This branch:  
Challenge dispute ‹ Disputes ‹ Payments ‹ Accused Hoopoe — WooCommerce


### Dispute Evidence Form

`develop` branch:  
5 9 0-dev – Dispute Evidence Form


This branch:  
After – Dispute Evidence Form





</details>

**Comment by mirka** [comment]
> * I tried to remove the types dependency, but we cannot use the built-in types. It complains about a missing declaration file. I searched on Slack and saw similar discussions. We need to keep using the DefinitelyTypes types.

FYI, built-in types are available from `@wordpress/components` 23.8.0 and above.

**Pull request merged**